exchange/exchange.go

```diff
@@ -236,6 +238,27 @@ func (e *exchange) HoldAuction(ctx context.Context, r AuctionRequest, debugLog *
 		}
 	}
 
+	generalWarningKey := openrtb_ext.BidderName(openrtb_ext.BidderReservedGeneral)
+
+	if !r.Account.DebugAllow && requestDebugInfo {
+		accountDebugDisabledWarning := openrtb_ext.ExtBidderMessage{
+			Code:    errortypes.AccountLevelDebugDisabledWarningCode,
+			Message: "debug turned off for account",
+		}
+		bidResponseExt.Warnings[generalWarningKey] = append(bidResponseExt.Warnings[generalWarningKey], accountDebugDisabledWarning)
+	}
+
+	if len(r.Warnings) != 0 {
+		for _, warning := range r.Warnings {
+			typedWarning := warning.(*errortypes.Warning)
```

Comment:
We shouldn't code against a specific struct here. Please just rely on the interfaces:
```
generalWarning := openrtb_ext.ExtBidderMessage{
  Code:    errortypes.ReadCode(warning),
  Message: warning.Error()
}
```

Reply:
Good point, fixed!